### ISSUE TITLE
10.0.0-beta.1 Incorrect DOM Ordering

### ISSUE THREAD

**Issue by davezuko** [preactjs/preact#1635]
Been loving 10.x.x thus far, thanks for all of your hard work! I did stumble on something today, however:

When a component renders two sibling elements (**A** which is dynamic and **B** which is static), preact produces DOM in the incorrect order if **A** renders `null` before later rendering content. I don't think it matters, but in case it does: the static sibling (**B**) does not have to be a component, i.e. you can replace `<B />` with the literal `<p>B</p>` and the issue will still repro.

Some details:
* This issue does _not_ repro on any **8.x.x** version. It _does_ repro on every **10.x.x** version, including alphas and betas.
* I tested the dynamic component using both hooks and classes, and the issue occurs in both flavors.

```js
/** @jsx h */
import { h, render } from "preact";
import { useState, useEffect } from "preact/hooks";

const A = () => {
  const [show, setShow] = useState(false);
  useEffect(() => {
    setTimeout(() => setShow(true), 1000);
  }, []);

  // intentionally render <div> instead of <p> to ensure the root elements
  // of both components are different, in case that changes any behavior.
  // The issue of course repros even if a <p> is used.
  return show ? <div>A</div> : null; 
};
const B = () => <p>B</p>;

render(
  <div>
    <A />
    <B />
  </div>,
  document.body
);

// Expected:
<div>
  <div>A</div>
  <p>B</p>
</div>

// Actual:
<div>
  <p>B</p>
  <div>A</div>
</div>
```

Link to sandboxes with minimal repro:
* Using classes: https://codesandbox.io/s/preactorderingissue-xtjs3
* Using hooks (same as above snippet): https://codesandbox.io/s/preactorderingissuehooks-s62ch

**Comment by davezuko** [comment]
This may be a duplicate of https://github.com/developit/preact/issues/1632. If so, please feel free to close.

**Comment by JoviDeCroock** [comment]
This does indeed feel closely related to #1632 but let's leave this open as an additional testing scenario.

Thanks for giving us an easy to reproduce example, we greatly appreciate it (even additionally checking classes and hooks). Thank you so much for that!

We'll look into this as soon as we can.

**Comment by marvinhagemeister** [comment]
This is fixed with #1689 and #1700 :tada: :100: We'll cut a new release soon sometime this week :checkered_flag:
